tools/engine_tool/lib/src/proc_utils.dart

```diff
@@ -0,0 +1,209 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:path/path.dart' as p;
+import 'package:process_runner/process_runner.dart';
+
+import 'environment.dart';
+import 'logger.dart';
+
+/// Returns a path to a JSON file that includes the logs
+/// and exit code of a process.
+String saveArtifacts(Environment env, ProcessRunnerResult result) {
+  final Directory systemTemp = Directory.systemTemp;
+  final String prefix = result.pid != null ? 'et_${result.pid}' : 'et';
+  final Directory artifacts = systemTemp.createTempSync(prefix);
+  final File resultFile = File(p.join(artifacts.path, 'result.json'));
+
+  final Map<String, Object> data = <String, Object>{};
+  if (result.pid != null) {
+    data['pid'] = result.pid!;
+  }
+  data['exitCode'] = result.exitCode;
+  data['stdout'] = result.stdout;
+  data['stderr'] = result.stderr;
+  // TODO(johnmccutchan): It would be nice to capture the command line and cwd
```

Comment:
Seems easy to me, I'd just do it, TODOs be damned

Reply:
Done.